### PR TITLE
Remove leftover code that used to handle ml4 files.

### PR DESCRIPTION
**Pull request by ppedrot**
In the build system, mostly.

### CODE REVIEW

**Review by ejgallego — APPROVED** [review]
I seem to recall that some of these cleanups are already inside some other PR, but honestly I've lost track of PRs at this point.

**Comment by ejgallego** [comment]
Will wait for @gares OK before merging.

**Review by gares — COMMENTED** [review]
I think it would be more polite to "parse" an .ml4 file name and issue an error there, mentioning the transition to .mlg

**Comment by ejgallego** [comment]
Where exactly @gares ?

**Comment by gares** [comment]
In coqprojectfile

File: lib/coqProject_file.ml [hunk 1]
```diff
@@ -220,7 +218,6 @@ let process_cmd_line ~warning_fn orig_dir proj args =
           | ".v" ->
             { proj with v_files = proj.v_files @ [sourced f] }
         | ".ml" -> { proj with ml_files = proj.ml_files @ [sourced f] }
-        | ".ml4" -> { proj with ml4_files = proj.ml4_files @ [sourced f] }
```
Comment (gares) [inline lib/coqProject_file.ml]: Here

**Comment by ejgallego** [comment]
@ppedrot I'll let you do the change then.

**Comment by ppedrot** [comment]
I've pushed a change with a specific error message. We need to remove this line by 8.11 though.

**Comment by ejgallego** [comment]
Thanks @ppedrot , merging later today.

**Pull request merged**